addon/-private/functional/modifier.ts

```diff
@@ -54,14 +60,72 @@ type Teardown = () => unknown;
 export default function modifier<
   E extends Element,
   P extends unknown[],
-  N extends Record<string, unknown>
+  N extends object
 >(
   fn: (element: E, positional: P, named: N) => void | Teardown
 ): FunctionBasedModifier<{
   Args: { Named: N; Positional: P };
   Element: E;
 }>;
 
+/**
+ * An API for writing simple modifiers.
+ *
+ * This function runs the first time when the element the modifier was applied
+ * to is inserted into the DOM, and it *autotracks* while running. Any values
+ * that it accesses will be tracked, including the arguments it receives, and if
+ * any of them changes, the function will run again.
```

Comment:
Since this is the docs for `eager: true` side, should this also mention that all of the incoming arguments are also eagerly consumed and will force the function to run again?

Reply:
Good call, yes.